SU-HSDO/suhumsci
# STN-621  Color band component

# READY FOR REVIEW

## Summary
As a site editor (Site Manager, Contributor), I want to add a large paragraph with a background color and large heading text, so that I can visually define sub-sections of content on the page in an appealing way. 

Campus Examples:
https://music.stanford.edu/what-we-do/instrumental-performance
https://humanrights.stanford.edu/publications-videos/overview

Jira Card: https://sparkbox.atlassian.net/browse/STN-621

## Need Review By (Date)
5/13

## Urgency
medium

## Steps to Test
1. npm run test
2. To see the option for the Color Band paragraph component, you will need to import the configuration using `lando drush @sparkbox_sandbox.local config-import --partial`
3. Create a Flexible Page and add a Color Band Paragraph component within the Components section of the page regions.
4. Feel free to create a Color Band with each color and also please use different amounts of text in each example you add.
- You should see these options: 'default' - `primary-hero-overlay`, 'cardinal` - global cardinal variable color added within this work and 'gray' - the global `gray-dark` variable (which was updated from the basic gray mentioned in the Jira card for accessibility concerns.
6. You should see a character limitation countdown in the plain text area field for Color Band as well, which should limit your text to 255 characters total.

**Verifications / Scenarios:**
- [x] Verify that the default color option for the themes is using the `primary-hero-overlay` color.
- [x] Verify that other utility classes, such as `hb-raised-cards` do not affect the layout or styling of the Color Bands.
- [x] Verify that the color band can be used with other paragraph components and they do not negatively interact with each other.
- [x] Verify that if you edit the layout of a page with color bands using the Layout tab, the Color Bands stay within their region and do not overlap over to the sidebars in the Layout Builder admin area when previewing content checkbox is checked.
- [x] Verify all colors and text of color bands color contrasts pass accessibility expectations, on both themes and all color pairings.
- [x] Verify the text within the Color Band is an H2.
- [x] Verify there is a margin between Color Band and other components so they are not right on top of one another.
- [x] Verify the Color Band component is not available within Rows.

## Add a Sidebar:
1. Edit the Layout of the page you've created using Layout and add a sidebar component, I generally grab the test-menu or user account menu as blocks to add.  I'll add them a couple of times to give me a decent size sidebar.
2. After you've added a single sidebar please verify the items below, please repeat these verifications by adding a second sidebar and/or removing a left sidebar and keeping a right sidebar only.
_**Note:** There was a `hb-sidebar` mixin added for this work to give the sidebars a white background and give them a z-index of 1 so they appear in front of the Color Band component._

**Sidebar Verifications:**
- [x] Verify the color band wraps the whole page, but goes behind the sidebar on either side or both.
- [x] Verify the text remains centered with the actual main page column.
- [x] Verify on mobile these items stack correctly and color band does not cause problems with mobile display.
- [x] Verify when you are looking at the Layout admin page using the Layout tab, that the sidebars are defaulting back to a background color of transparent and z-index: auto on the Drupal admin theme area, as to not cause possible issues with the styling added from the mixin.

![image](https://user-images.githubusercontent.com/9355698/118051705-71ebd000-b34f-11eb-8b34-a7b533d4c2f5.png)

![image](https://user-images.githubusercontent.com/9355698/118051942-e292ec80-b34f-11eb-8eeb-aae4740c43b5.png)


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
